provider-service/base/pkg/streams/sinks/webhook_sink_unit_test.go

```diff
@@ -0,0 +1,91 @@
+//go:build unit
+
+package sinks
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	resty "github.com/go-resty/resty/v2"
+	"github.com/jarcoal/httpmock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/sky-uk/kfp-operator/argo/common"
+	. "github.com/sky-uk/kfp-operator/provider-service/base/pkg"
+	"go.uber.org/zap/zapcore"
+	"testing"
+)
+
+func TestSinksUnitSuite(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Sinks Unit Suite")
+}
+
+var _ = Context("SendEvents", func() {
+	var logger, _ = common.NewLogger(zapcore.DebugLevel)
+	var ctx = logr.NewContext(context.Background(), logger)
+	runCompletionEventData := common.RunCompletionEventData{
+		Status:                "",
+		PipelineName:          common.NamespacedName{},
+		RunConfigurationName:  nil,
+		RunName:               nil,
+		RunId:                 "",
+		ServingModelArtifacts: nil,
+		PipelineComponents:    nil,
+		Provider:              "",
+	}
+
+	When("webhook sink receives StreamMessage", func() {
```

Comment:
This test setup should probably follow:

```go
When("webhook sink receives a valid StreamMessage", func() {
        It("sends RunCompletionEventData to the webhook successfully", func() { ... })
})```

```go
When("webhook fails to send a StreamMessage", func() {
        It("triggers the message OnFailureHandler", func() { ... })
})```

Reply:
changed 847762891ac9e4c53e697be2fbd16d03bfb70260